buffer/src/main/java/io/netty5/buffer/api/ComponentIterator.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.buffer.api;
+
+import io.netty5.buffer.api.ComponentIterator.Next;
+
+public interface ComponentIterator<T extends Next> extends AutoCloseable {
```

Comment:
Add javadocs

Reply:
There are multiple things that need to have their javadocs expanded, and also tests needs to be added. I'm working on that.